test_suites/packageupgrade/packageupgrade_test.go

```diff
@@ -0,0 +1,132 @@
+package packageupgrade
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/cloud-image-tests/utils"
+)
+
+const (
+	googet      = "C:\\ProgramData\\GooGet\\googet.exe"
+	repoPath    = "C:\\ProgramData\\GooGet\\repos\\google-compute-engine-testing.repo"
+	repoContent = `"- name: google-compute-engine-testing
+  url: https://packages.cloud.google.com/yuck/repos/google-compute-engine-testing
+  useoauth: true"`
+)
+
+func ChangeRepo(t *testing.T) {
+	command := fmt.Sprintf("cmd.exe /c del /Q C:\\ProgramData\\GooGet\\repos\\*")
+	utils.FailOnPowershellFail(command, "Error deleting stable repo", t)
+
+	command = fmt.Sprintf("%s available", googet)
+	err := utils.CheckPowershellReturnCode(command, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	command = fmt.Sprintf("Set-Content -Path %s -Value %s", repoPath, repoContent)
+	_, err = utils.RunPowershellCmd(command)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestDriverUpgrade(t *testing.T) {
+	utils.WindowsOnly(t)
+	ChangeRepo(t)
+	drivers := []string{
+		"google-compute-engine-driver-pvpanic",
+		"google-compute-engine-driver-gga",
+		"google-compute-engine-driver-balloon",
+		"google-compute-engine-driver-gvnic",
+		/*
+			The driver packages will need to be updated to accomodate this test behavior.
+			We want to uncomment the following drivers once they are fixed,
+			"google-compute-engine-driver-netkvm",
+			"google-compute-engine-driver-vioscsi",
+		*/
+	}
+
+	for _, driver := range drivers {
+		command := fmt.Sprintf("%s installed %s", googet, driver)
+		output, err := utils.RunPowershellCmd(command)
+		if err != nil {
+			t.Fatalf("Error getting package status for '%s'", driver)
+		}
+		inString := fmt.Sprintf("No package matching filter \"%s\" installed.", driver)
+		if !strings.Contains(output.Stdout, inString) {
+			command := fmt.Sprintf("%s -noconfirm install -reinstall %s", googet, driver)
+			output, err := utils.RunPowershellCmd(command)
+			if err != nil {
+				t.Fatalf("Error reinstalling '%s': %v", driver, err)
+			}
+			reString := fmt.Sprintf("Reinstallation of %s.* completed", driver)
+			matched, err := regexp.MatchString(reString, output.Stdout)
+			if !matched {
+				t.Fatalf("Reinstall of '%s' returned unexpected result: %s", driver, output.Stdout)
+			}
+		} else {
+			command := fmt.Sprintf("%s -noconfirm install %s", googet, driver)
+			output, err := utils.RunPowershellCmd(command)
+			if err != nil {
+				t.Fatalf("Error installing '%s': %v", driver, err)
+			}
+			reString := fmt.Sprintf("Installation of %s.* and all dependencies completed", driver)
+			matched, err := regexp.MatchString(reString, output.Stdout)
+			if !matched {
+				t.Fatalf("Install of '%s' returned unexpected result: %s", driver, output.Stdout)
+			}
+		}
+	}
+}
+
+func TestPackageUpgrade(t *testing.T) {
+	utils.WindowsOnly(t)
+	ChangeRepo(t)
+	packages := []string{
+		"certgen",
+		"googet",
+		//"google-cloud-ops-agent",
```

Comment:
Why is this one commented out?

Reply:
ops agent uses its own repo which is not included in testing; I need to figure out in a future change the best way to test that repo